### PR TITLE
Remove validation when adding or updating consumer

### PR DESCRIPTION
**Pull request by scottf**
1. Do not require name or durable when adding or updating a consumer.
2. Do not reject when inactive threshold is set and a durable is set.

### CODE REVIEW

File: src/test/java/io/nats/client/impl/JetStreamManagementTests.java [hunk 1]
```diff
@@ -654,12 +654,19 @@ public void testAddDeleteConsumer() throws Exception {
             assertTrue(iae.getMessage().contains("Stream cannot be null or empty"));
             iae = assertThrows(IllegalArgumentException.class, () -> jsm.addOrUpdateConsumer(STREAM, null));
             assertTrue(iae.getMessage().contains("Config cannot be null"));
-            iae = assertThrows(IllegalArgumentException.class, () -> jsm.addOrUpdateConsumer(STREAM, cc));
-            assertTrue(iae.getMessage().contains("Durable cannot be null"));
```
Comment (scottf) [inline src/test/java/io/nats/client/impl/JetStreamManagementTests.java:658]: No longer validating

File: src/test/java/io/nats/client/impl/JetStreamManagementTests.java [hunk 2]
```diff
@@ -654,12 +654,19 @@ public void testAddDeleteConsumer() throws Exception {
...
 
+            // durable and name can both be null
+            ConsumerInfo ci = jsm.addOrUpdateConsumer(STREAM, cc);
+            assertNotNull(ci.getName());
```
Comment (scottf) [inline src/test/java/io/nats/client/impl/JetStreamManagementTests.java:660]: test that the server gives a name when one is not supplied

File: src/test/java/io/nats/client/impl/JetStreamManagementTests.java [hunk 3]
```diff
@@ -654,12 +654,19 @@ public void testAddDeleteConsumer() throws Exception {
...
+
+            // threshold can be set for durable
             final ConsumerConfiguration cc2 = ConsumerConfiguration.builder().durable(DURABLE).inactiveThreshold(10000).build();
-            iae = assertThrows(IllegalArgumentException.class, () -> jsm.addOrUpdateConsumer(STREAM, cc2));
-            assertTrue(iae.getMessage().contains("Inactive Threshold"));
+            ci = jsm.addOrUpdateConsumer(STREAM, cc2);
+            assertEquals(10000, ci.getConsumerConfiguration().getInactiveThreshold().toMillis());
```
Comment (scottf) [inline src/test/java/io/nats/client/impl/JetStreamManagementTests.java:665]: test that an inactiveThreshold is allowed when a durable name is supplied

File: src/main/java/io/nats/client/impl/NatsJetStreamManagement.java [hunk 4]
```diff
@@ -125,11 +124,6 @@ public PurgeResponse purgeStream(String streamName, PurgeOptions options) throws
     public ConsumerInfo addOrUpdateConsumer(String streamName, ConsumerConfiguration config) throws IOException, JetStreamApiException {
         validateStreamName(streamName, true);
         validateNotNull(config, "Config");
-        validateNotNull(config.getDurable(), "Durable"); // durable name is required when creating consumers
```
Comment (scottf) [inline src/main/java/io/nats/client/impl/NatsJetStreamManagement.java:128]: durable not required

File: src/main/java/io/nats/client/impl/NatsJetStreamManagement.java [hunk 5]
```diff
@@ -125,11 +124,6 @@ public PurgeResponse purgeStream(String streamName, PurgeOptions options) throws
...
-        Duration d = config.getInactiveThreshold();
-        if (d != null && !d.isZero()) {
-            throw new IllegalArgumentException("Durable consumers cannot have an Inactive Threshold.");
-        }
```
Comment (scottf) [inline src/main/java/io/nats/client/impl/NatsJetStreamManagement.java:132]: inactiveThreshold allowed even when there is a durable

**Review by piotrpio — COMMENTED** [review]
Looks good, just one nit

**Review by piotrpio — APPROVED** [review]
LGTM!

**Pull request merged**